### PR TITLE
Remove all dependencies on Guava

### PR DESCRIPTION
**Pull request by Peilin-Yang**

### CODE REVIEW

File: src/main/java/io/anserini/ltr/TwitterFeatureExtractor.java [hunk 1]
```diff
@@ -102,10 +106,12 @@ protected Analyzer getAnalyzer() {
 
   @Override
   protected Set<String> getFieldsToLoad() {
-    return Sets.newHashSet(getIdField(), getTermVectorField(),
-            StatusField.FOLLOWERS_COUNT.name,
-            StatusField.FRIENDS_COUNT.name,
-            StatusField.IN_REPLY_TO_STATUS_ID.name);
+    return Stream.of(
```
Comment (lintool) [inline src/main/java/io/anserini/ltr/TwitterFeatureExtractor.java]: Is this more clear than something like?

> `Set<String> h = new HashSet<>(Arrays.asList("a", "b"));`

I actually like above better...

File: src/main/java/io/anserini/ltr/feature/FeatureExtractor.java [hunk 2]
```diff
@@ -16,28 +17,29 @@
...
-          .put("UniqueQueryTerms", UniqueTermCount.class)
-          .put("UnorderedSequentialPairs", UnorderedSequentialPairsFeatureExtractor.class)
-          .put("OrderedSequentialPairs", OrderedSequentialPairsFeatureExtractor.class)
-          .put("TwitterHashtagCount", HashtagCount.class)
-          .put("IsTweetReply", IsTweetReply.class)
-          .put("TwitterLinkCount", LinkCount.class)
-          .put("TwitterFollowerCount", TwitterFollowerCount.class)
-          .put("TwitterFriendCount", TwitterFriendCount.class).build();
+  final Map<String, Class<?>> EXTRACTOR_MAP = new HashMap<String, Class<?>>() {{
+    put("AvgICTF", AvgICTFFeatureExtractor.class);
```
Comment (lintool) [inline src/main/java/io/anserini/ltr/feature/FeatureExtractor.java]: Can use HashMap<>()?

File: src/main/java/io/anserini/rerank/lib/AxiomReranker.java [hunk 3]
```diff
@@ -117,7 +114,7 @@ public AxiomReranker(String field, SearchArgs args) throws IOException {
 
   @Override
   public ScoredDocuments rerank(ScoredDocuments docs, RerankerContext<T> context) {
-    Preconditions.checkState(docs.documents.length == docs.scores.length);
+    assert (docs.documents.length == docs.scores.length);
```
Comment (lintool) [inline src/main/java/io/anserini/rerank/lib/AxiomReranker.java]: no space, right? below also.

File: src/main/java/io/anserini/ltr/feature/FeatureExtractor.java (hunk 2 above)
Comment (Peilin-Yang) [reply src/main/java/io/anserini/ltr/feature/FeatureExtractor.java]: Cannot use diamond constructor with anonymous inner class....

**Pull request merged**